datafusion/src/physical_optimizer/pruning.rs

```diff
@@ -1394,7 +1394,7 @@ mod tests {
         let expr = col("b1").not().eq(lit(true));
         let p = PruningPredicate::try_new(&expr, schema).unwrap();
         let result = p.prune(&statistics).unwrap();
-        assert_eq!(result, vec![true, false, false, true, true]);
+        assert_eq!(result, vec![true, true, false, true, true]);
```

Comment:
interesting, looks like arrow2 fixed a bug that exists in arrow-rs?

Reply:
I didn't get too much background on this, since this is a test against pruning of `a != true`, I think the current behavior is expected